### PR TITLE
Worktree: Provide ability to add excludes to worktree

### PR DESCRIPTION
**Pull request by maguro**
Adding excludes to the worktree seemed to be the best course of action since it minimizes
the amount of code changes; neatly sidestepping the issue of ballooning constructors.

It also seems to be philosophically in line w/ other code.  For example, there is no code to
obtain the username and email from the ~/.gitconfig file and yet those two pieces of
information are needed for commits, etc.

With that said, convenience functions, LoadGlobalPatterns() and LoadGlobalPatterns(),
are provided to obtain any excludes that may be declared in ~/.gitconfig and /etc/gitconfig
respectively.

Signed-off-by: Alan D. Cabrera <adc@toolazydogs.com>

### CODE REVIEW

**Comment by maguro** [comment]
This is for issue #760.

**Comment by maguro** [comment]
Not sure why the appveyor job fails when travis-ci passes.

File: worktree.go [hunk 1]
```diff
@@ -34,6 +35,9 @@ type Worktree struct {
 	// Filesystem underlying filesystem.
 	Filesystem billy.Filesystem
 
+	// External excludes not found in the repository .gitignore
```
Comment (mcuadros) [inline worktree.go]: can you remove the extra line?

Comment (maguro) [reply worktree.go]: done

File: plumbing/format/gitignore/dir_test.go [hunk 2]
```diff
@@ -2,19 +2,29 @@ package gitignore
...
 
 	. "gopkg.in/check.v1"
 	"gopkg.in/src-d/go-billy.v4"
 	"gopkg.in/src-d/go-billy.v4/memfs"
 )
 
 type MatcherSuite struct {
-	FS billy.Filesystem
+	GFS  billy.Filesystem // git repository root
+
```
[the remark this answers is not in the document]
Comment (maguro) [reply plumbing/format/gitignore/dir_test.go]: done

**Pull request merged**